### ISSUE TITLE
Cannot create entry containing a date field

### ISSUE THREAD

**Issue by g-div** [strapi/strapi#1860]
**Informations**
- **Node.js version**: v10.1.0
- **npm version**: 6.0.1
- **Strapi version**: 3.0.0-alpha.14
- **Database**: MongoDB
- **Operating system**: Linux

**What is the current behavior?**
Creating a new entry containing a *date* field returns the error:
```
CastError: Cast to date failed for value "o" at path "date"
```
which is otherwise not easily debuggable because the error gets catched [here](https://github.com/strapi/strapi/blob/60a7e99d11a2ac15317c7aeb378dec5022fb7776/packages/strapi-plugin-content-manager/config/queries/mongoose.js#L111) displaying: 
```
This Cast,to,date,failed,for,value,o,at,path,date is already taken
```

**Steps to reproduce the problem**
- Create a new *Type* using the *Content Type Builder*
- Add a *date* field
- Create a new *Entry* of the new *Type*
- Fill out the input fields
- Click on *Save*

**What is the expected behavior?**
The new entry gets created

**Suggested solutions**
Maybe this error printed in the console will help further:
```
Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.
```

**Comment by g-div** [comment]
Is #1855 related to this issue ?

**Comment by lauriejim** [comment]
Hello @g-div! I'm not able to reproduce this issue.

What is your attribute name ?
Did you migrate your app or it's a fresh new one ?

**Comment by g-div** [comment]
It's a new fresh app and the attribute name is `date`.
I've also just tried to reproduce this from scratch and was unable to, but I understood that the problem comes from this part of the model definition:
```
    "date": {
      "default": "o",
      "type": "date"
},
```

I'm not exactly sure where the `o` as default value is coming from, but I'm pretty sure that I've just created the model from the UI, without modifying the file myself. Even if it was my mistake accidentally writing an `o` as default value, I think the main problem is [this](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-content-manager/config/queries/mongoose.js#L111) error catching which isn't helpful enough and maybe in such cases the server start should be interrupted when default values can't be validated.
